src/modules/learner_module.py

```diff
@@ -56,17 +56,16 @@ def fit(self, training_loader, validation_loader,
             epochs=1, 
             early_stopping=False,
             sample_weights=None,
-            verbose=0):
+            verbose=0,
+            early_stopping_patience = 5,
+            early_stopping_counter = 0,
```

Comment:
This should not be a parameter and should be initialised right before the epoch loop. Sorry for the confusion. 
Otherwise everything looks good